pkg/cmd/pulumi/deployment.go

```diff
@@ -82,7 +85,110 @@ func newDeploymentSettingsCmd() *cobra.Command {
 		}),
 	}
 
+	cmd.AddCommand(newDeploymentSettingsInitCmd())
 	cmd.AddCommand(newDeploymentSettingsPullCmd())
+	cmd.AddCommand(newDeploymentSettingsUpdateCmd())
+	cmd.AddCommand(newDeploymentSettingsDestroyCmd())
+
+	return cmd
+}
+
+func newDeploymentSettingsInitCmd() *cobra.Command {
+	var stack string
+
+	cmd := &cobra.Command{
+		Use:        "init",
+		SuggestFor: []string{"new", "create"},
+		Args:       cmdutil.ExactArgs(0),
+		Short:      "Initializes the stack deployment settings file",
+		Long:       "",
+		Run: cmdutil.RunFunc(func(cmd *cobra.Command, args []string) error {
+			ctx := cmd.Context()
+			interactive := cmdutil.Interactive()
+
+			displayOpts := display.Options{
+				Color:         cmdutil.GetGlobalColorization(),
+				IsInteractive: interactive,
+			}
+
+			project, _, err := readProject()
+			if err != nil && !errors.Is(err, workspace.ErrProjectNotFound) {
+				return err
+			}
+
+			currentBe, err := currentBackend(ctx, project, displayOpts)
+			if err != nil {
+				return err
+			}
+
+			if !currentBe.SupportsDeployments() {
+				return fmt.Errorf("backends of this type %q do not support deployments",
+					currentBe.Name())
+			}
+
+			s, err := requireStack(ctx, stack, stackOfferNew|stackSetCurrent, displayOpts)
+			if err != nil {
+				return err
+			}
+
+			wd, err := os.Getwd()
+			if err != nil {
+				return err
+			}
+			repo, err := git.PlainOpen(wd)
+			if err != nil {
+				return err
+			}
+
+			newStackDeployment := &workspace.ProjectStackDeployment{
+				DeploymentSettings: apitype.DeploymentSettings{
+					SourceContext: &apitype.SourceContext{
+						Git: &apitype.SourceContextGit{
+							RepoDir: ".",
```

Comment:
Is the `RepoDir` necessary here?

Reply:
No. I initially wanted to infer it but it is tricky, so I went with a sensitive default in order to make it easier for customers to change from the file. Ideally this would change when we convert the init to a wizard